osu.Game.Rulesets.Mania/Mods/ManiaModPlayfieldCover.cs

```diff
@@ -27,22 +35,50 @@ public abstract class ManiaModPlayfieldCover : ModHidden, IApplicableToDrawableR
         [SettingSource("Coverage", "The proportion of playfield height that notes will be hidden for.")]
         public abstract BindableNumber<float> Coverage { get; }
 
+        [SettingSource("Change coverage based on combo", "Increase the coverage as combo increases.")]
+        public BindableBool ComboBasedCoverage => new BindableBool(true);
+
+        protected readonly BindableInt Combo = new BindableInt();
+
+        public override void ApplyToScoreProcessor(ScoreProcessor scoreProcessor)
+        {
+            Combo.BindTo(scoreProcessor.Combo);
+
+            // Default value of ScoreProcessor's Rank in Hidden/Fadein Mod should be SS+
+            scoreProcessor.Rank.Value = ScoreRank.XH;
+        }
+
         public virtual void ApplyToDrawableRuleset(DrawableRuleset<ManiaHitObject> drawableRuleset)
         {
             ManiaPlayfield maniaPlayfield = (ManiaPlayfield)drawableRuleset.Playfield;
 
+            if (ComboBasedCoverage.Value)
+            {
+                coveringWrappers = new List<PlayfieldCoveringWrapper>();
+                clock = maniaPlayfield.Clock;
+            }
```

Comment:
Why are these values being assigned conditional on `ComboBasedCoverage`? It's not only unnecessary, but it also makes the nullability annotations on the fields actively misleading (because both can *very feasibly* be `null`).